random/random.mbti

```diff
@@ -3,13 +3,20 @@ package moonbitlang/core/random
 // Values
 
 // Types and methods
-type RandomState
-impl RandomState {
-  gen_double(Self) -> Double
-  gen_int(Self) -> Int
-  gen_int64(Self) -> Int64
-  init_state(~seed : Int = ..) -> Self
-  new() -> Self
+type Rand
+impl Rand {
+  double64(Self) -> Double
+  float32(Self) -> Float
+  int(Self) -> Int
+  int64(Self) -> Int64
+  int64N(Self, Int64) -> Int64
+  intN(Self, Int) -> Int
```

Comment:
Use optional argument for `intN`? for example `int(Self, ~limit?:Int)`

Reply:
uuml avoid allocation -- imperative API